src/core/path.c

```diff
@@ -518,10 +533,23 @@ static void path_enter_running(Path *p) {
                 return;
         }
 
-        r = manager_add_job(UNIT(p)->manager, JOB_START, trigger, JOB_REPLACE, NULL, &error, NULL);
+        details = activation_details_new(UNIT(p));
+        if (!details) {
+                r = -ENOMEM;
+                goto fail;
+        }
+
+        if (free_and_strdup(&(ACTIVATION_DETAILS_PATH(details))->trigger_path_filename, trigger_path) < 0) {
+                r = -ENOMEM;
```

Comment:
as mentioned earlier, i'd just do:

```c
r = free_and_strdup();
if (r < 0)
        return r;
```

i.e. if we have error codes, always propagate them as we have them